Mathlib/Analysis/LocallyConvex/Basic.lean

```diff
@@ -199,18 +195,24 @@ theorem Balanced.smul_mem_mono [SMulCommClass 𝕝 𝕜 E] (hs : Balanced 𝕝 s
         exact div_le_one_of_le₀ hba (norm_nonneg _)
       (a⁻¹ • b) • a • x = b • x := by rw [smul_comm, smul_assoc, smul_inv_smul₀ ha₀]
 
-theorem Balanced.subset_smul (hA : Balanced 𝕜 A) (ha : 1 ≤ ‖a‖) : A ⊆ a • A := by
-  rw [← @norm_one 𝕜] at ha; simpa using hA.smul_mono ha
+theorem Balanced.subset_smul (hs : Balanced 𝕜 s) (ha : 1 ≤ ‖a‖) : s ⊆ a • s := by
+  rw [← @norm_one 𝕜] at ha; simpa using hs.smul_mono ha
 
-theorem Balanced.smul_congr (hs : Balanced 𝕜 A) (h : ‖a‖ = ‖b‖) : a • A = b • A :=
+theorem Balanced.smul_congr (hs : Balanced 𝕜 s) (h : ‖a‖ = ‖b‖) : a • s = b • s :=
   (hs.smul_mono h.le).antisymm (hs.smul_mono h.ge)
 
-theorem Balanced.smul_eq (hA : Balanced 𝕜 A) (ha : ‖a‖ = 1) : a • A = A :=
-  (hA _ ha.le).antisymm <| hA.subset_smul ha.ge
+theorem Balanced.smul_eq (hs : Balanced 𝕜 s) (ha : ‖a‖ = 1) : a • s = s :=
+  (hs _ ha.le).antisymm <| hs.subset_smul ha.ge
 
 /-- A balanced set absorbs itself. -/
-theorem Balanced.absorbs_self (hA : Balanced 𝕜 A) : Absorbs 𝕜 A A :=
-  .of_norm ⟨1, fun _ => hA.subset_smul⟩
+theorem Balanced.absorbs_self (hs : Balanced 𝕜 s) : Absorbs 𝕜 s s :=
+  .of_norm ⟨1, fun _ => hs.subset_smul⟩
+
+end NormedDivisionRing
+
+section NormedField
+
+variable [NormedField 𝕜] [AddCommGroup E] [Module 𝕜 E] {s A : Set E} {x : E} {a b : 𝕜}
 
 theorem Balanced.smul_mem_iff (hs : Balanced 𝕜 s) (h : ‖a‖ = ‖b‖) : a • x ∈ s ↔ b • x ∈ s :=
   ⟨(hs.smul_mem_mono · h.ge), (hs.smul_mem_mono · h.le)⟩
```

Comment:
It should be possible to generalize this one too, but it'll require a different version of `smul_mem_mono` without `SMulCommClass` assumption (or with a different TC assumption instead of `SMulCommClass`). It's definitely true in case `𝕜 = 𝕝`, because we can use `b * a⁻¹` instead of `a⁻¹ • b` in the original proof.

Reply:
You'd need `SmullCommClass 𝕜 (MulOpposite 𝕜) E` as an assumption instead, which is awkward because we are missing many right actions. We could have both versions if that's helpful.

Reply:
AFAICT, we don't have a `NormedRing` instance for `MulOpposite`. I don't want to spend more time on this generalization, so I'm going to just merge this PR. We can generalize more lemmas later.